pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BacklogQuotaManager.java

```diff
@@ -53,9 +53,10 @@
 
     public BacklogQuotaManager(PulsarService pulsar) {
         this.isTopicLevelPoliciesEnable = pulsar.getConfiguration().isTopicLevelPoliciesEnabled();
+        double backlogQuotaGB = pulsar.getConfiguration().getBacklogQuotaDefaultLimitGB();
         this.defaultQuota = BacklogQuotaImpl.builder()
-                .limitSize(pulsar.getConfiguration().getBacklogQuotaDefaultLimitGB()
-                        * BacklogQuotaImpl.BYTES_IN_GIGABYTE)
+                .limitSize(backlogQuotaGB > 0 ? (long) backlogQuotaGB * BacklogQuotaImpl.BYTES_IN_GIGABYTE
```

Comment:
@rdhabalia / @eolivelli  shouldn't this be `(long) (backlogQuotaGB * BacklogQuotaImpl.BYTES_IN_GIGABYTE)`? Because here the double value of user-configured backlog in GB is converted to long before converting to bytes which keeps the same behaviour as before. Aren't we losing the purpose of using double here? Or I am misunderstood? 

e.g:
```
public class HelloWorld {
    public static void main(String[] args) {
        double a = 1.6;
        System.out.println((long) a * 1024); // 1024, unexpected
        System.out.println((long) (a * 1024)); // 1638, expected
    }
}
```

Reply:
good catch !
can you please send a PR to fix it ASAP ?

Reply:
here it is: https://github.com/apache/pulsar/pull/11697